modules/developer-cli-odo-switching-between-ephemeral-and-persistent-storage.adoc

```diff
@@ -0,0 +1,68 @@
+// Module included in the following assemblies:
+//
+// *cli_reference/developer_cli_odo/creating_and_deploying_applications_with_odo/working-with-storage.adoc
+
+[id="switching-between-ephemeral-and-persistent-storage_{context}"]
+= Switching between ephemeral and persistent storage
+
+You can switch between ephemeral and persistent storage in your project by using the `odo preference` command. `odo preference` modifies the global preference in your cluster.
+
+When persistent storage is enabled, the cluster stores the information between the restarts. 
+
+When ephemeral storage is enabled, the cluster does not store the information between the restarts.
+
+Ephemeral storage is enabled by default. 
+
+.Procedure
+
+. See the preference currently set in your project:
++
+[source,terminal]
+----
+odo preference view
+----
++
+.Example output
++
+[source,terminal]
+----
+PARAMETER             CURRENT_VALUE
+UpdateNotification
+NamePrefix
+Timeout
+BuildTimeout
+PushTimeout
+Experimental
+PushTarget
+Ephemeral             true
+----
+
+. To unset the ephemeral storage and set the persistent storage:
++
+[source,terminal]
+----
+$ odo preference set Ephemeral false
+----
+
+. To set the ephemeral storage again:
++
+[source,terminal]
+----
+$ odo preference set ephemeral
```

Comment:
Running `odo preference set ephemeral` after doing `odo preference set Ephemeral false`, as mentioned in a step earlier, is going to throw an error. It's going to fail with `✗  please provide a parameter name and value`.

To move from using Persistent storage (which can be enabled *only* by doing `odo preference set Ephemeral false`) to using Ephemeral storage, one has to do either `odo preference unset Ephemeral` (because when you unset its value, odo assumes its default value which is `true`), or `odo preference set Ephemeral true` (so that you don't leave any room for confusion about what odo's going to read as far as Ephemeral preference's setting is concerned.)